libraries/lib-music-information-retrieval/tests/TatumQuantizationFitBenchmarking.cpp

```diff
@@ -0,0 +1,220 @@
+      std::back_inserter(samples), [&](const std::string& wavFile) {
+         const WavMirAudioReader audio { wavFile };
+         UpdateHash(audio, hash);
+         QuantizationFitDebugOutput debugOutput;
+         std::function<void(double)> progressCb;
+         GetMusicalMeterFromSignal(audio, tolerance, progressCb, &debugOutput);
+         ProgressBar(progressBarWidth, 100 * count++ / numFiles);
+         const auto expected = GetBpmFromFilename(wavFile);
+         const auto truth = expected.has_value();
+         const std::optional<OctaveError> error =
+            truth ?
+               std::make_optional(GetOctaveError(*expected, debugOutput.bpm)) :
+               std::nullopt;
+         sampleValueCsv << (truth ? "true" : "false") << ","
+                        << debugOutput.score << ","
+                        << 60. * debugOutput.tatumQuantization.numDivisions /
+                              debugOutput.audioFileDuration
+                        << "," << debugOutput.bpm << ","
+                        << ToString(debugOutput.timeSignature) << ","
+                        << (error.has_value() ? error->factor : 0.) << ","
+                        << (error.has_value() ? error->remainder : 0.) << ","
+                        << debugOutput.tatumQuantization.lag << "," << wavFile
+                        << "\n";
+         return Sample { truth, debugOutput.score, error };
+      });
+
+   // AUC of ROC curve. Tells how good our loop/not-loop clasifier is.
+   const auto rocInfo = GetRocInfo(
+      samples, loopClassifierSettings.at(tolerance).allowedFalsePositiveRate);
+
+   const auto strictThreshold =
+      GetRocInfo(
+         samples, loopClassifierSettings.at(FalsePositiveTolerance::Strict)
+                     .allowedFalsePositiveRate)
+         .threshold;
+
+   // Get RMS of octave errors. Tells how good the BPM estimation is.
+   const auto octaveErrors = std::accumulate(
+      samples.begin(), samples.end(), std::vector<double> {},
+      [&](std::vector<double> octaveErrors, const Sample& sample) {
+         if (sample.octaveError.has_value())
+            octaveErrors.push_back(sample.octaveError->remainder);
+         return octaveErrors;
+      });
+   const auto octaveErrorStd = std::sqrt(
+      std::accumulate(
+         octaveErrors.begin(), octaveErrors.end(), 0.,
+         [&](double sum, double octaveError) {
+            return sum + octaveError * octaveError;
+         }) /
+      octaveErrors.size());
+
+   std::ofstream summaryFile {
+      std::string(CMAKE_CURRENT_SOURCE_DIR) +
+      "/GetMusicalMeterFromSignalBenchmarkingOutput/summary.txt"
+   };
+
+   summaryFile << std::setprecision(std::numeric_limits<double>::digits10 + 1)
+               << "AUC: " << rocInfo.areaUnderCurve << "\n"
+               << "Strict Threshold (Minutes-and-Seconds): " << strictThreshold
+               << "\n"
+               << "Lenient Threshold (Beats-and-Measures): "
```

Comment:
From @Paul-Licameli:
> I notice the checked-in summary.txt was not generated by this version of the .cpp file

Reply:
Yes, the commits I'm preparing should fix this. Most importantly, the calculated thresholds are consistent with the values used in `MusicInformationRetrieval.h`,
```
static const std::unordered_map<FalsePositiveTolerance, LoopClassifierSettings>
   loopClassifierSettings {
      { FalsePositiveTolerance::Strict, { .01, 0.8367879995455243 } },
      { FalsePositiveTolerance::Lenient, { .1, 0.7336582045210642 } },
   };
   ```